cpp/src/IceLocatorDiscovery/PluginI.cpp

```diff
@@ -52,40 +51,43 @@ class Request : public std::enable_shared_from_this<Request>
     const Ice::OperationMode _mode;
     const Ice::Context _context;
     const Ice::ByteSeq _inParams;
-    AMDCallback _amdCB;
+    function<void(bool, const pair<const uint8_t*, const uint8_t*>)> _responseCallback;
+    function<void(exception_ptr)> _exceptionCallback;
     exception_ptr _exception;
 
-    Ice::LocatorPrxPtr _locatorPrx;
+    optional<Ice::LocatorPrx> _locatorPrx;
```

Comment:
I have to keep this as optional because it is not initialized in the constructor.